mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1891,25 +2095,27 @@ static func::FuncOp createCpuGemmKernelWithMlir(ModuleOp module,
   Location loc = module->getLoc();
 
   SmallVector<Type, 3> cpuTypes = params.types;
-  // Raise floats to f32
-  for (Type &type : cpuTypes) {
-    if (type.isa<FloatType>())
-      type = b.getF32Type();
-  }
-
   SmallVector<Type, 3> argTypes;
   getGemmTypes(cpuTypes, argTypes, /*isCpuVerifier=*/true);
 
   constexpr llvm::StringLiteral cpuKernName("host_naive_gemm");
   auto func =
       b.create<func::FuncOp>(loc, cpuKernName, b.getFunctionType(argTypes, {}));
+  module.push_back(func);
```

Comment:
Move this up from the end, because emitMemcpy() traces back to the ModuleOp from inside the function.  Tried to make it consistent around the file.